src/app/pages/PerpetualPage/components/TradeForm/index.tsx

```diff
@@ -44,18 +47,41 @@ export const TradeForm: React.FC<ITradeFormProps> = ({
   const { t } = useTranslation();
   const { checkMaintenance, States } = useMaintenance();
   const inMaintenance = checkMaintenance(States.PERPETUAL_TRADES);
+  const ammState = usePerpetual_queryAmmState();
+  const perpParameters = usePerpetual_queryPerpParameters();
+
+  const [lotSize, lotPrecision] = useMemo(() => {
+    const lotSize = Number(perpParameters.fLotSizeBC.toPrecision(8));
+    const lotPrecision = lotSize.toString().replace(/^.*\.?/, '').length;
+    return [lotSize, lotPrecision];
+  }, [perpParameters.fLotSizeBC]);
+
+  const maxTradeSize = useMemo(
+    () =>
+      Number(
+        Math.abs(
+          getMaximalTradeSizeInPerpetual(
+            0,
+            trade.position === TradingPosition.LONG ? 1 : -1,
+            ammState,
+            perpParameters,
+          ),
+        ).toPrecision(8),
+      ),
+    [trade.position, ammState, perpParameters],
+  );
 
   const [amount, setAmount] = useState(fromWei(trade.amount));
   const onChangeOrderAmount = useCallback(
     (amount: string) => {
-      const roundedAmount = bignumber(amount || '0')
-        .dividedBy(STEP_SIZE)
-        .round()
-        .mul(STEP_SIZE);
+      const roundedAmount = shrinkToLot(
```

Comment:
I am not sure about using helper functions from `temporaryUtils`, we will eventually switch to an npm package. But we can leave it like that for now.